platform/packages/platform/src/reply.rs

```diff
@@ -3,59 +3,83 @@
 //! Here are defined wrappers for deserializing such structures.
 
 use prost::Message;
-use serde::{de::DeserializeOwned, Deserialize};
+use serde::de::DeserializeOwned;
 
-use sdk::cosmwasm_std::{from_json, Addr, Api, Binary, Reply, StdError, StdResult};
+use sdk::{
+    cosmos_sdk_proto::cosmwasm::wasm::v1::{
+        MsgExecuteContractResponse, MsgInstantiateContract2Response, MsgInstantiateContractResponse,
+    },
+    cosmwasm_std::{from_json, Addr, Api, Binary, Reply, StdError, StdResult},
+};
 
 pub struct InstantiateResponse<T> {
     pub address: Addr,
-    pub data: Option<T>,
+    pub data: T,
 }
 
-pub fn from_instantiate<T>(api: &dyn Api, reply: Reply) -> StdResult<InstantiateResponse<T>>
-where
-    T: DeserializeOwned,
-{
-    #[derive(Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub address: Vec<u8>,
-        #[prost(bytes, tag = "2")]
-        pub data: Vec<u8>,
+impl InstantiateResponse<Vec<u8>> {
+    fn into_addr(self) -> Addr {
+        self.address
     }
+}
 
-    let response = decode::<ReplyData>(reply)?;
+pub fn from_instantiate_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+    from_instantiate_inner::<MsgInstantiateContractResponse>(api, reply)
+        .map(InstantiateResponse::into_addr)
+}
 
-    Ok(InstantiateResponse {
-        address: api
-            .addr_validate(&String::from_utf8(response.address).map_err(|_| {
-                StdError::generic_err("Address field contains invalid UTF-8 data!")
-            })?)?,
-        data: maybe_from_json(response.data)?,
-    })
+pub fn from_instantiate2_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+    from_instantiate_inner::<MsgInstantiateContract2Response>(api, reply)
+        .map(InstantiateResponse::into_addr)
 }
 
 pub fn from_execute<T>(reply: Reply) -> StdResult<Option<T>>
 where
     T: DeserializeOwned,
 {
-    #[derive(prost::Message)]
-    struct ReplyData {
-        #[prost(bytes, tag = "1")]
-        pub data: Vec<u8>,
+    decode::<MsgExecuteContractResponse>(reply)
+        .map(|data| data.data)
+        .map(Binary)
+        .and_then(from_json)
+}
+
+struct UncheckedInstantiateResponse {
+    address: String,
+    data: Vec<u8>,
+}
+
+impl From<MsgInstantiateContractResponse> for UncheckedInstantiateResponse {
+    fn from(
+        MsgInstantiateContractResponse { address, data }: MsgInstantiateContractResponse,
+    ) -> Self {
+        Self { address, data }
     }
+}
 
-    decode::<ReplyData>(reply)
-        .map(|data| data.data)
-        .and_then(maybe_from_json)
+impl From<MsgInstantiateContract2Response> for UncheckedInstantiateResponse {
+    fn from(
+        MsgInstantiateContract2Response { address, data }: MsgInstantiateContract2Response,
+    ) -> Self {
+        Self { address, data }
+    }
+}
+
```

Comment:
Added extra complexity handling `data`.  A simple trait a-la `Address` implemented for the two structures MsgInstantiateContractResponse and MsgInstantiate2ContractResponse would suffice